spec/services/export/s3_downloader_spec.rb

```diff
@@ -0,0 +1,82 @@
+require "rails_helper"
+
+RSpec.describe Export::S3Downloader do
+  let(:response_body) { double("response_body", read: double) }
+  let(:response) { double("response", body: response_body) }
+  let(:aws_credentials) { double("aws credentials") }
+  let(:aws_client) { instance_double(Aws::S3::Client, get_object: response) }
+
+  let(:s3_bucket) { double("s3 bucket") }
+  let(:s3_uploader_config) {
+    instance_double(
+      Export::S3UploaderConfig,
+      key_id: "key id",
+      secret_key: "secret key",
+      region: "region",
+      bucket: s3_bucket
+    )
+  }
+
+  let(:filename) { "Q1_report0101202312345.csv" }
+
+  let(:public) { true }
+
+  subject {
+    Export::S3Downloader.new(filename: filename)
+  }
+
+  before do
+    allow(Aws::Credentials).to receive(:new).and_return(aws_credentials)
+    allow(Aws::S3::Client).to receive(:new).and_return(aws_client)
+    allow(Export::S3UploaderConfig).to receive(:new).and_return(s3_uploader_config)
+  end
+
+  describe "#initialize" do
+    context "when instantiating the Aws::S3::Client" do
+      it "sets credentials: using an Aws::Credentials object" do
+        subject
+
+        expect(Aws::Credentials).to have_received(:new).with(
+          s3_uploader_config.key_id,
+          s3_uploader_config.secret_key
+        )
+        expect(Aws::S3::Client).to have_received(:new).with(hash_including(
+          credentials: aws_credentials
+        ))
+      end
+
+      it "sets region: using the region from the S3UploaderConfig" do
```

Comment:
Is there a reason for the colons in the test descriptions?

Reply:
I suppose  it's a sort of shorthand way of saying "sets the region key...". Do you think that would be a bit clearer?

Reply:
Ah got you, that or backticks maybe?